### ISSUE TITLE
extra_vars not propagated to syntax check when resolving import_playbook: references

### ISSUE THREAD

**Issue by rodrigosaito** [ansible/ansible-lint#5042]
##### Summary

`ansible-lint`'s `extra_vars` (declared in `.ansible-lint.yml`) are honored when linting a playbook directly, but they are **not** honored when the same playbook is reached transitively via `import_playbook:` from another playbook.

In the transitive case, `ansible-lint` logs `ERROR  Failed to load <name> playbook due to failing syntax check.` and silently skips linting the imported playbook. The lint summary still says `Passed: 0 failure(s)` and the process exits 0, which makes the failure easy to miss in CI.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console
$ ansible-lint --version
ansible-lint 26.4.0 using ansible-core:2.18.16 ansible-compat:26.3.0 ruamel-yaml:0.19.1 ruamel-yaml-clib:0.2.15
```

- ansible installation method: pip
- ansible-lint installation method: pip
- OS: macOS 14 / Linux (reproduces on both)
- Python: 3.11

##### STEPS TO REPRODUCE

`.ansible-lint.yml`
```yaml
extra_vars:
  my_host: localhost
```

`playbooks/inner.yml`
```yaml
---
- name: Inner play
  hosts: "{{ my_host }}"
  gather_facts: false
  tasks:
    - name: Noop
      ansible.builtin.debug:
        msg: hi
```

`playbooks/outer.yml `
```yaml
---
- name: Import inner
  ansible.builtin.import_playbook: inner.yml
```

Then run both lint commands and compare:

```console
$ ansible-lint --offline playbooks/inner.yml
Passed: 0 failure(s), 0 warning(s) in 1 files processed of 1 encountered. Last profile that met the validation criteria was 'production'.
$ echo $?
0

$ ansible-lint --offline playbooks/outer.yml
ERROR    Failed to load inner.yml playbook due to failing syntax check.
Passed: 0 failure(s), 0 warning(s) in 1 files processed of 1 encountered. Last profile that met the validation criteria was 'production'.
$ echo $?
0
```

The same `inner.yml` is lintable on its own, but is silently dropped from linting (and not counted as a failure) when reached via `import_playbook:`, even though `extra_vars` is configured.

##### Desired Behavior

A playbook should lint the same way whether I run `ansible-lint` on it directly or on another playbook that imports it. Concretely, given the reproducer above I expect:

```console
$ ansible-lint --offline playbooks/outer.yml
Passed: 0 failure(s), 0 warning(s) in 2 files processed of 2 encountered.
$ echo $?
0
```

— the same result I get from `ansible-lint --offline playbooks/inner.yml`, because the `extra_vars` from `.ansible-lint.yml` should apply to both.

##### Actual Behavior

`extra_vars` from `.ansible-lint.yml` are applied when `inner.yml` is linted directly, but not when it is reached via `import_playbook:` from `outer.yml`. In the latter case the lint output is:

```
ERROR    Failed to load inner.yml playbook due to failing syntax check.
Passed: 0 failure(s), 0 warning(s) in 1 files processed of 1 encountered.
```

and the process exits 0 — so a CI run that lints the entry-point playbook will not catch the failure.
